File: lab1/Event.cpp
#include "Event.h"
#include "EventType.h"

Event::Event(double arrivalTime, EventType type){
  this->arrivalTime = arrivalTime;
  this->type = type;
}

double Event::getArrivalTime(){
  return this->arrivalTime;
}

EventType Event::getType(){
  return this->type;
}

Event * Event::minEvent(Event * a, Event * b){
  Event * min;
  double at = a->getArrivalTime();
  double bt = b->getArrivalTime();
  if(at < bt){
    min = a;
  }else{
    min = b;
  }
  return min;

}
Event * Event::minEvent(Event * a, Event * b, Event * c){
  Event * min;
  if(a->getArrivalTime() < b->getArrivalTime()){
    min = a;
  }else{
    min = b;
  }

  if(min->getArrivalTime() <= c->getArrivalTime()){
    return min;
  }

  return c;
}

File: lab2/src/lab2/event/Event.java
package lab2.event;

/**
 * Models an event for simulation
 * @author kolive
 *
 */
public class Event {

	private EventType type;
	private double time;
	private int sn;
	private boolean error;
	
	
	public Event(EventType type, double time){
		this.type = type;
		this.time = time;
	}
	
	public Event(EventType type, double time, int sn, boolean error){
		this.type = type;
		this.time = time;
		this.error = error;
		this.sn = sn;
	}
	
	public double getTime(){
		return time;
	}
	
	public EventType getType(){
		return type;
	}
	
	public boolean isError(){
		if(type == EventType.TO) return false;
		return error;
	}
	
	public int getSN(){
		if(type == EventType.TO) return -1;
		return sn;
	}
	
	
	
}
